pallets/creditcoin/src/types.rs

```diff
@@ -581,8 +581,310 @@ pub enum TaskData<AccountId, Balance, BlockNum, Hash, Moment> {
+	}
+
+	#[test]
+	fn test_blockchain_as_bytes() {
+		let bitcoin = Blockchain::Bitcoin;
+		assert_eq!(bitcoin.as_bytes(), b"bitcoin");
+
+		let other = Blockchain::Other(BoundedVec::try_from(b"my-awesome-chain".to_vec()).unwrap());
+		assert_eq!(other.as_bytes(), b"my-awesome-chain");
+	}
+
+	#[test]
+	fn test_orderid_to_hex() {
+		mock::ExtBuilder::default().build_and_execute(|| {
+			let order_id = OrderId::Deal(TestInfo::new_defaults().create_deal_order().1);
+			let expected = [
+				49, 102, 48, 53, 53, 56, 100, 48, 99, 99, 50, 54, 97, 99, 99, 57, 51, 52, 102, 101,
+				100, 99, 99, 57, 54, 57, 99, 102, 51, 56, 54, 101, 52, 56, 100, 57, 49, 99, 102,
+				50, 50, 55, 56, 51, 98, 55, 54, 97, 49, 57, 56, 98, 100, 101, 55, 52, 97, 97, 48,
+				48, 52, 101, 102, 56,
+			];
+			assert_eq!(order_id.to_hex(), expected);
+		})
+	}
+
+	#[test]
+	fn test_orderid_expiration() {
+		mock::ExtBuilder::default().build_and_execute(|| {
+			let (_, deal_order_id) = TestInfo::new_defaults().create_deal_order();
+			let order_id = OrderId::Deal(deal_order_id.clone());
+			assert_eq!(order_id.expiration(), deal_order_id.expiration());
+		})
+	}
+
+	#[test]
+	fn test_legacysighash_try_from_when_string_is_shorter_than_60_chars() {
+		let result = LegacySighash::try_from("too-short");
+		assert!(result.is_err());
+	}
+
+	#[test]
+	fn test_legacysighash_try_from_when_string_is_longer_than_60_chars() {
+		let result = LegacySighash::try_from(
+			"this-dummy-string-is-very-very-very-long-and-cannot-be-a-legacy-sighash",
+		);
+		assert!(result.is_err());
+	}
+
+	#[test]
+	fn test_legacysighash_serialize_deserialize() {
+		let value = LegacySighash::default();
+
+		let json_string = serde_json::to_string(&value).unwrap();
+		let deserialized_value = serde_json::from_str(&json_string).unwrap();
+		assert_eq!(value, deserialized_value);
+	}
+
+	#[test]
+	fn test_duration_new() {
+		let result = Duration::new(5u64, 4000000u32);
+		assert_eq!(result, Duration::from_millis(5004));
+	}
+
+	#[test]
+	fn test_bidterms_match_with() {
+		mock::ExtBuilder::default().build_and_execute(|| {
+			let loan_terms = TestInfo::new_defaults().loan_terms;
+			let ask_terms = AskTerms::try_from(loan_terms.clone()).unwrap();
+			let bid_terms = BidTerms::try_from(loan_terms).unwrap();
+
+			assert!(bid_terms.match_with(&ask_terms));
+		})
+	}
+
+	#[test]
+	fn test_bidterms_agreed_terms() {
+		mock::ExtBuilder::default().build_and_execute(|| {
+			let loan_terms = TestInfo::new_defaults().loan_terms;
+			let ask_terms = AskTerms::try_from(loan_terms.clone()).unwrap();
+			let bid_terms = BidTerms::try_from(loan_terms).unwrap();
+
+			assert_eq!(
+				ask_terms.agreed_terms(bid_terms.clone()),
+				bid_terms.agreed_terms(&ask_terms),
+			);
+		})
+	}
+
+	#[test]
+	fn test_evmchainid_new() {
+		assert_eq!(EvmChainId::new(46), EvmChainId::from(46),);
+	}
+
+	#[test]
+	#[allow(clippy::clone_on_copy)]
+	fn test_invalid_term_length_error_clone_and_runtime_debug() {
```

Comment:
not really testing anything.
```suggestion
	fn exercise_invalid_term_length_error_clone_and_runtime_debug() {
```